pkgs/by-name/ex/ext3grep/package.nix

```
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchurl
+, e2fsprogs
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ext3grep";
+  version = "0.10.2";
+
+  src = fetchurl {
+    url = "https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/ext3grep/ext3grep-${finalAttrs.version}.tar.gz";
+    hash = "sha256-WG8+k50v/XgvbwBrgaPfLcR3xtoD8h7biGDFxPcZjz4=";
+  };
+
+  nativeBuildInputs = [ e2fsprogs ];
+
+  patches = [
+    ./ext3grep-build.patch
+    ./ext3grep-e2fsprogs-1.44.1.patch
+  ];
+
+  buildPhase = ''
+    ./configure --prefix=$out
+    make
+  '';
+
+  meta = with lib; {
+    description = "A tool to investigate an ext3 file system for deleted content and possibly recover it";
+    mainProgram = "ext3grep";
+    homepage = "https://code.google.com/archive/p/ext3grep/";
+    maintainers = with maintainers; [ d3vil0p3r ];
+    platforms = platforms.unix;
```

```suggestion
    platforms = platforms.unix;
```
the darwin builds are failing -- should this be narrowed to `platforms.linux` ? perhaps the debian patches for clang would fix the build.

---

Currently I don't have a darwin machine. Can you test it by applying that clang patch file, so, if it works, I can add it to the list of file patches?

---

you can apply the patches and see if the CI succeeds or fails.

---

Oh true. I am by smartphone and I didn't see the ofborg `
ext3grep, ext3grep.passthru.tests on <arch>-darwin` failing. Once I back home, I apply the required changes ^^.

---

ok -- it looks like the [failures](https://logs.ofborg.org/?key=nixos/nixpkgs.288134&attempt_id=e407ad98-9edc-40cb-8598-d9334114e489) on darwin

```console
In file included from custom.cc:28:
./ext3.h:107:40: error: use of undeclared identifier 'i_uid_low'; did you mean 'uid_low'?
    __u16 uid_low(void) const { return i_uid_low; }
                                       ^~~~~~~~~
                                       uid_low
./ext3.h:107:11: note: 'uid_low' declared here
    __u16 uid_low(void) const { return i_uid_low; }
          ^
./ext3.h:107:40: error: reference to non-static member function must be called; did you mean to call it with no arguments?
    __u16 uid_low(void) const { return i_uid_low; }
                                       ^~~~~~~~~
[...]
```

 are due to defines in `ext2_fs.h` in the `e2fsprogs` package and they are ifdefed for linux or kernel only
```c
#if defined(__KERNEL__) || defined(__linux__)
#define i_reserved1     osd1.linux1.l_i_reserved1
#define i_frag          osd2.linux2.l_i_frag
#define i_fsize         osd2.linux2.l_i_fsize
#define i_uid_low       i_uid
#define i_gid_low       i_gid
#define i_uid_high      osd2.linux2.l_i_uid_high
#define i_gid_high      osd2.linux2.l_i_gid_high
```

so this should just be `platforms.linux`